File: src-tauri/src/models/relation_obj.model.rs
/* sys lib */
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum TypesField {
  OneToOne,
  OneToMany,
  ManyToOne,
  ManyToMany,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RelationObj {
  pub nameTable: String,
  pub typeField: TypesField,
  pub nameField: String,
  pub newNameField: String,
  pub relations: Option<Vec<RelationObj>>,
}

File: src/app/models/relation-obj.model.ts
export enum TypesField {
  OneToOne = "OneToOne",
  OneToMany = "OneToMany",
  ManyToOne = "ManyToOne",
  ManyToMany = "ManyToMany",
}

export interface RelationObj {
  nameTable: string;
  typeField: TypesField;
  nameField: string;
  newNameField: string;
  targetField?: string; // New: Column to join against in target table
  relations: RelationObj[] | null;
}
